long-ride-alerts/src/main/java/org/apache/flink/training/exercises/longrides/LongRidesExercise.java

```diff
@@ -97,18 +99,61 @@ public static void main(String[] args) throws Exception {
 
     @VisibleForTesting
     public static class AlertFunction extends KeyedProcessFunction<Long, TaxiRide, Long> {
-
+        private ValueState<TaxiRide> rideState;
```

Comment:
Since this class implement `Serializable` interface
It has to be `transient`
` private transient ValueState<TaxiRide> rideState;`
Also I recommend to specify
`serialVersionUID`
private static final long serialVersionUID = 1L;